.github/workflows/ca-clone-sequential-test.yml

```diff
@@ -1429,6 +1429,774 @@ jobs:
 
           diff expected actual
 
+      ####################################################################################################
+      # Switch cert request ID generator to newLegacy and verify if serials
+      # have gaps when range is updated
+      #
+      # It should work like the legacy but with correct range.
+      - name: Switch primary to newLegacy
+        run: |
+          # Update CS.cfg
+          docker exec primaryds ldapsearch \
+              -H ldap://primaryds.example.com:3389 \
+              -D "cn=Directory Manager" \
+              -w Secret.123 \
+              -b ou=certificateRepository,ou=ranges,dc=ca,dc=pki,dc=example,dc=com \
+              -s one \
+              -o ldif_wrap=no \
+              -LLL '(&(host=primary.example.com)(SecurePort=8443))' beginRange | tee output
+
+          # Since there is a range the value are retrieved from the range. In general, id range is not
+          # present the value from CS.cfg are used.
+          BEGIN_SERIAL=0x$(cat output | sed -n 's/beginRange: \(.*\)/\1/p' | sort -n | tail -1)
+          END_SERIAL=$(printf "0x%x\n" $(( BEGIN_SERIAL + 0x12 -1 )) )
+
+          docker exec primary pki-server ca-config-set dbs.beginSerialNumber $BEGIN_SERIAL
+          docker exec primary pki-server ca-config-set dbs.endSerialNumber $END_SERIAL
+
+          docker exec primary pki-server ca-config-set dbs.serialIncrement 0x12
+          docker exec primary pki-server ca-config-set dbs.serialLowWaterMark 0x9
+          docker exec primary pki-server ca-config-set dbs.serialCloneTransferNumber 0x9
+          docker exec primary pki-server ca-config-set dbs.request.id.generator newLegacy
+
+          docker exec primary pki-server ca-config-set dbs.cert.id.generator newLegacy
+
+          # Update new_range in DS
+          docker exec primaryds ldapsearch \
+              -H ldap://primaryds.example.com:3389 \
+              -D "cn=Directory Manager" \
+              -w Secret.123 \
+              -b ou=certificateRepository,ou=ranges,dc=ca,dc=pki,dc=example,dc=com \
+              -s one \
+              -o ldif_wrap=no \
+              -LLL endRange | tee output              
+
+          END_RANGE=0x$(cat output | sed -n 's/endRange: \(.*\)/\1/p' | sort -n | tail -1)
+          NEW_NEXT=$((END_RANGE + 1))
+          docker exec -i primaryds ldapmodify \
+              -H ldap://primaryds.example.com:3389 \
+              -D "cn=Directory Manager" \
+              -w Secret.123 << EOF
+          dn: ou=certificateRepository,ou=ca,dc=ca,dc=pki,dc=example,dc=com
+          changeType: modify
+          replace: nextRange
+          nextRange: $NEW_NEXT
+          EOF
```

Comment:
This probably should be identical to `Repository.getNextRange()` too, but in `getNextRange()` it allocates a new range in addition to updating the `nextRange`. If we don't allocate a new range here would it create a gap?

Reply:
As in the comment before, the change should not modify the ranges but fix the values so it is different from the method where the update is done because of a new range

Reply:
I guess the question is, how will other servers that have not been upgraded yet see the new `nextRange`, and how will they be affected by it? If it works like a regular new range allocation then we don't need to worry about it, but this one seems to be different.

Reply:
`nextRange` is shared among all the clones. The migration should be all or nothing. It has not to be synced but before requesting a new range.

Reply:
The problem is only if a clone is asking for a new range while the master has already migrated. This could create problems although I have not tested yet. I'll test to verify if it possible to handle or not.

Reply:
OK, we require all servers to be shutdown, so this is not a problem.